src/moparser.js

```diff
@@ -70,48 +65,51 @@ Parser.prototype._checkMagick = function () {
 };
 
 /**
- * Read the original strings and translations from the input MO file. Use the
- * first translation string in the file as the header.
+ * Read the original strings and translations from the input MO file.
+ * Use the first translation string in the file as the header.
  */
 Parser.prototype._loadTranslationTable = function () {
-  let offsetOriginals = this._offsetOriginals;
-  let offsetTranslations = this._offsetTranslations;
+  let offsetOriginals = this._offsetOriginals || 0;
+  let offsetTranslations = this._offsetTranslations || 0;
   let position;
   let length;
   let msgid;
   let msgstr;
 
-  for (let i = 0; i < this._total; i++) {
-    // msgid string
-    length = this._fileContents[this._readFunc](offsetOriginals);
-    offsetOriginals += 4;
-    position = this._fileContents[this._readFunc](offsetOriginals);
-    offsetOriginals += 4;
-    msgid = this._fileContents.subarray(
-      position,
-      position + length,
-    );
-
-    // matching msgstr
-    length = this._fileContents[this._readFunc](offsetTranslations);
-    offsetTranslations += 4;
-    position = this._fileContents[this._readFunc](offsetTranslations);
-    offsetTranslations += 4;
-    msgstr = this._fileContents.subarray(
-      position,
-      position + length,
-    );
-
-    if (!i && !msgid.toString()) {
-      this._handleCharset(msgstr);
+  if (this._total) {
+    for (let i = 0; i < this._total; i++) {
+      if (this._fileContents === null) continue;
+      // msgid string
+      length = this._fileContents.readUInt32LE(offsetOriginals);
+      offsetOriginals += 4;
+      position = this._fileContents.readUInt32LE(offsetOriginals);
+      offsetOriginals += 4;
+      msgid = this._fileContents.subarray(
+        position,
+        position + length
+      );
+
+      // matching msgstr
+      length = this._fileContents.readUInt32LE(offsetTranslations);
+      offsetTranslations += 4;
+      position = this._fileContents.readUInt32LE(offsetTranslations);
```

Comment:
If I'm reading this correctly you have removed the logic to read and write `BE`, if the magic byte is `BE` its looks like these should use the `BE` functions.

Reply:
https://www.gnu.org/software/gettext/manual/gettext.html#index-_002d_002dendianness_002c-msgfmt-option

Reply:
This is where I wouldn't mess with the read function logic. The parser does use it, and currently supports parsing both big and little endian mo files.

There are much better ways of doing this, but I wouldn't fix it here.